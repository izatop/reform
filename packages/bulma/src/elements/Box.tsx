import React from "react";
import {XProps} from "../interfaces";
import {MakeProps} from "../type";
import {ElementFactory} from "../utils";

const config = ElementFactory.create({component: "box"});
export const Box = config.factory<MakeProps, XProps<"div">>(({props, children}) => {
    return <div {...props}>{children}</div>;
});
