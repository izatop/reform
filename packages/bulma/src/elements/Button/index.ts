export * from "./Button";
export * from "./Buttons";
