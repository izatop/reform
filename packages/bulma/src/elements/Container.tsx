import React from "react";
import {XProps} from "../interfaces";
import {MakeBreakpoint} from "../type";
import {ElementFactory} from "../utils";

interface IContainer {
    fluid?: boolean;
}

const config = ElementFactory.create({component: "container"});
export const Container = config.factory<MakeBreakpoint<IContainer>, XProps<"div">>(({props, children}) => (
    <div {...props}>{children}</div>
));
