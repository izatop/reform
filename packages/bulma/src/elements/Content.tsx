import React from "react";
import {MakeProps, XProps} from "../interfaces";
import {ConfigFactory} from "../utils";

const config = ConfigFactory.create({component: "content"});
export const Content = config.factory<MakeProps, XProps<"div">>(({props, children}) => (
    <div {...props}>{children}</div>));
