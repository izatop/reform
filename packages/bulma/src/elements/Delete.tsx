import * as React from "react";
import {XProps} from "../interfaces";
import {IsSize} from "../props";
import {MakeProps} from "../type";
import {ElementFactory} from "../utils";

const config = ElementFactory.create({component: "delete"});
export const Delete = config.factory<MakeProps<IsSize>, XProps<"a">>(({props, children}) => {
    return <a {...props}>{children}</a>;
});
