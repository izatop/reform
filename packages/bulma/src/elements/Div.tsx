import React from "react";
import {XProps} from "../interfaces";
import {MakeProps} from "../type";
import {ElementFactory} from "../utils";

const config = ElementFactory.create({displayName: "div"});
export const Div = config.factory<MakeProps, XProps<"div">>(({props, children}) => (
    <div {...props}>{children}</div>
));
