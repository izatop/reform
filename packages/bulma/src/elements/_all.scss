@charset "utf-8";

@import "Button/all";
@import "Icon/all";
@import "Figure/all";
@import "Notification/all";
@import "Progress/all";
@import "Table/all";
@import "Tag/all";
@import "Title/all";
@import "Box.scss";
@import "Container.scss";
@import "Content.scss";
@import "Delete.scss";
