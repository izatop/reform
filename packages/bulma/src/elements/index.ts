export * from "./Button";
export * from "./Icon";
export * from "./Figure";
export * from "./Notification";
export * from "./Progress";
export * from "./Table";
export * from "./Tag";
export * from "./Title";
export * from "./Box";
export * from "./Container";
export * from "./Content";
export * from "./Delete";
export * from "./Node";
