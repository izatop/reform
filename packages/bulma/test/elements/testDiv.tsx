import * as React from "react";
import * as renderer from "react-test-renderer";
import {Div} from "../../src/elements/Div";

test("Div", () => {
    const div = renderer.create(<Div textAlign={"centered"} textSize={6} className={"foo"}>Text</Div>);
    expect(div.toJSON()).toMatchSnapshot();
});
