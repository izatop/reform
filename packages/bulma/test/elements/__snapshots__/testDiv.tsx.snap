// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Div 1`] = `
<div
  className="has-text-centered is-size-6"
>
  Text
</div>
`;
